pandas/_libs/algos.pyx

```diff
@@ -840,7 +840,7 @@ def rank_1d(
     elif rank_t is float64_t:
         mask = np.isnan(values)
     elif rank_t is int64_t:
-        mask = values == NPY_NAT
+        mask = missing.isnaobj(values)
```

Comment:
this is going to be all-False

We probably need to specifically handle `datetimelike` case in which the existing check is correct